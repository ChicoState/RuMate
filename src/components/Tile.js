import React from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import firebase from 'firebase'
import * as Haptic from 'expo-haptics';

const Tile = ({ title, color, text, textColor, location, nav, params, run }) => {
  if (!params) {
    params = {}
  }
  return (
    <TouchableOpacity onPress={() => {
      if (run == 'logout') {
        firebase.auth().signOut()
        Alert("Logged Out")
      } else if (run == 'haptic-select'){
        Haptic.selectionAsync()
      }
      nav.navigate(location, params)
      }}>
      <View style={[{backgroundColor: color, borderColor: color}, styles.wrapper]}>
        <Text style={[styles.title, {color: textColor}]}>{title}</Text>
        <Text style={[{color: textColor}]}>{text}</Text>
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    paddingBottom: '5%',
    paddingTop: '2%'
  },
  wrapper: {
    marginHorizontal: 5,
    marginVertical: 3,
    paddingHorizontal: 10,
    borderRadius: 10,
    borderWidth: 1,
    height: 100,
    width: "95%",
    alignSelf: 'center'
  }
});

export default Tile;