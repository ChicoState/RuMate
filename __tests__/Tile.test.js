import React from 'react';
import renderer from 'react-test-renderer';
import firebase from 'firebase';
import { shallow, configure } from 'enzyme';
import Tile from '../src/components/Tile';
import Adapter from 'enzyme-adapter-react-16';
import firebaseConfig from './Firebase.test.config';

describe('<Tile />', () => {

  it('has 1 child', () => {
    const tree = renderer.create(<Tile />).toJSON();
    console.log(tree.children.length);
    expect(tree.children.length).toBe(1);
  });
  
  it('renders correctly', () => {
    const tree = renderer.create(<Tile />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  configure({ adapter: new Adapter() });

  test('test onPress functionality', () => {
    const navigation = { navigate: jest.fn() };
    const wrapper = shallow(<Tile nav={navigation}/>);
    console.log(wrapper.props().onPress());
  });

  test('test onPress Haptic', () => {
    const navigation = { navigate: jest.fn() };
    const wrapper = shallow(<Tile run="haptic-select" nav={navigation}/>);
    console.log(wrapper.props().onPress());
  });
  
  test('test onPress logout', () => {
    firebase.initializeApp(firebaseConfig);
    const navigation = { navigate: jest.fn() };
    const wrapper = shallow(<Tile run="logout" nav={navigation}/>);
    console.log(wrapper.props().onPress());
  });

});

