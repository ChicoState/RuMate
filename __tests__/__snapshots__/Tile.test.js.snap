// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": undefined,
          "borderColor": undefined,
        },
        Object {
          "alignSelf": "center",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 100,
          "marginHorizontal": 5,
          "marginVertical": 3,
          "paddingHorizontal": 10,
          "width": "95%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "paddingBottom": "5%",
            "paddingTop": "2%",
          },
          Object {
            "color": undefined,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": undefined,
          },
        ]
      }
    />
  </View>
</View>
`;
